tests/test_registration.py

```
@@ -0,0 +1,532 @@
+    # To make ChildCommandHandler handle SampleCommand, one would do:
+    # @command_handler(SampleCommand)
+    # class ChildCommandHandler(ParentCommandHandler): ...
+    # Then it would overwrite ParentCommandHandler due to reregistration.
+
+    assert default_handler_registry.get_command_handler(AnotherCommand) is ChildHandlesAnotherCommand
+
+# Test clearing a custom registry
+def test_custom_registry_clear():
+    custom_registry = HandlerRegistry()
+    @command_handler(SampleCommand, registry=custom_registry)
+    class CH(BaseSampleCommandHandler): pass
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)